### PR TITLE
Feature-users

### PR DESCRIPTION
**Pull request by rishabh-vinsol**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+class UsersController < ApplicationController
```
Comment (Sparsh-Goyal-2000) [inline app/controllers/users_controller.rb:1]: Add routes for users controller

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  before_action :set_user, only: %i[ show edit update destroy ]
+
+  def index
+    @users = User.all
+  end
+
+  def show
```
Comment (Sparsh-Goyal-2000) [inline app/controllers/users_controller.rb:8]: Create empty actions in application controller and overwrite in other controllers only if you need them

File: app/helpers/users_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module UsersHelper
```
Comment (Sparsh-Goyal-2000) [inline app/helpers/users_helper.rb:1]: Don't use scaffold to create all files.
Only create the required files for now.

File: app/views/users/_user.html.erb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    <%= user.first_name %>
+  </p>
+
+  <p>
+    <strong>Last name:</strong>
+    <%= user.last_name %>
+  </p>
+
+  <p>
+    <strong>Type:</strong>
```
Comment (Sparsh-Goyal-2000) [inline app/views/users/_user.html.erb:13]: this has been changed to role.
fix this

File: app/views/users/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,42 @@
+<%= form_with(model: user) do |form| %>
+  <% if user.errors.any? %>
```
Comment (Sparsh-Goyal-2000) [inline app/views/users/_form.html.erb:2]: create a partial to show errors.

File: app/views/users/_user.json.jbuilder [hunk 6]
```diff
@@ -0,0 +1,2 @@
+json.extract! user, :id, :first_name, :last_name, :type, :email, :password, :created_at, :updated_at
```
Comment (Sparsh-Goyal-2000) [inline app/views/users/_user.json.jbuilder:1]: When are we using this file ?
If not in use, them remove this.

File: app/views/users/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,14 @@
+<p style="color: green"><%= notice %></p>
```
Comment (Sparsh-Goyal-2000) [inline app/views/users/index.html.erb:1]: Create a partial to show errors or success messages

File: app/views/users/index.json.jbuilder [hunk 8]
```diff
@@ -0,0 +1 @@
+json.array! @users, partial: "users/user", as: :user
```
Comment (Sparsh-Goyal-2000) [inline app/views/users/index.json.jbuilder:1]: When are we using this file ?
If not in use, them remove this.

File: app/views/users/show.json.jbuilder [hunk 9]
```diff
@@ -0,0 +1 @@
+json.partial! "users/user", user: @user
```
Comment (Sparsh-Goyal-2000) [inline app/views/users/show.json.jbuilder:1]: same

File: app/views/users/_form.html.erb [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  </div>
+
+  <div>
+    <%= form.label :last_name, style: "display: block" %>
+    <%= form.text_field :last_name %>
+  </div>
+
+  <div>
+    <%= form.label :type, style: "display: block" %>
+    <%= form.number_field :type %>
```
Comment (Sparsh-Goyal-2000) [inline app/views/users/_form.html.erb:26]: make it select type from the available roles

File: app/controllers/users_controller.rb [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+  def edit
+  end
+
+  def create
+    @user = User.new(user_params)
+
+    respond_to do |format|
+      if @user.save
+        format.html { redirect_to user_url(@user), notice: "User was successfully created." }
+        format.json { render :show, status: :created, location: @user }
```
Comment (Sparsh-Goyal-2000) [inline app/controllers/users_controller.rb:24]: Why two formats?
Can we use single format here?

File: app/controllers/users_controller.rb [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @user.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def destroy
+    @user.destroy
+
+    respond_to do |format|
```
Comment (Sparsh-Goyal-2000) [inline app/controllers/users_controller.rb:47]: check for failure here as well.

File: app/controllers/users_controller.rb [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+    end
+  end
+
+  private
+    def set_user
+      @user = User.find_by(id: params[:id])
+    end
+
+    def user_params
+      params.require(:user).permit(:first_name, :last_name, :type, :email, :password)
```
Comment (Sparsh-Goyal-2000) [inline app/controllers/users_controller.rb:59]: type or role ?

File: app/views/users/_form.html.erb (hunk 10 above)
Comment (Sparsh-Goyal-2000) [reply app/views/users/_form.html.erb:26]: type or role

File: app/views/users/show.html.erb [hunk 14]
```diff
@@ -0,0 +1,10 @@
+<p style="color: green"><%= notice %></p>
```
Comment (Sparsh-Goyal-2000) [inline app/views/users/show.html.erb:1]: same

**Review by Sparsh-Goyal-2000 — CHANGES_REQUESTED** [review]

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (rishabh-vinsol) [reply app/controllers/users_controller.rb:8]: Shouldn't I just remove them?

**Pull request merged**